lib/parse.js

```diff
@@ -45,6 +46,14 @@ var parseValues = function parseQueryStringValues(str, options) {
     var skipIndex = -1; // Keep track of where the utf8 sentinel was found
     var i;
 
+    var updateKey = function updateObjKey(key, value) {
```

Comment:
any reason not to take `obj` as an argument?

Reply:
it's an inner function. 
if I take `obj` as a declared argument, it would be more reasonable to take the whole function out of `parseValues` function. 
I believe that it is more `close` to where it is actually called, but I can take it out.

Reply:
My bad, your original code phrase support set val with array. No need to strip anything out.  ：）